docs/intro/ink-vs-solidity.md

```diff
@@ -719,78 +719,102 @@ Now, to perform the cross-contract call:
+#![cfg_attr(not(feature = "std"), no_std, no_main)]
 
-// ...
+#[ink::contract]
+mod call_contract {
+    use ink::{
+        env::call::{
+            build_call, 
+            Call, 
+            ExecutionInput, 
+            Selector
+        },
+        prelude::vec::Vec,
+    };
 
-// see: https://github.com/paritytech/ink-examples/blob/main/multisig/lib.rs#L535
-fn invoke_transaction(
-    &mut self,
-    callee: AccountId,
-    transfer_amount: u128,
-    function_selector: [u8; 4],
-    transaction_data: Vec<u8>,
-    gas_limit: u64) -> Result<()> {
-
-    let result = build_call::<<Self as ::ink_lang::reflect::ContractEnv>::Env>()
-        .call_type(
-            Call::new()
-                .callee(callee) // contract to call
-                .gas_limit(*gas_limit)
-                .transferred_value(transfer_amount), // value to transfer with call
-        )
-        .exec_input(
-            ExecutionInput::new(Selector::from(*function_selector))
-                    .push_arg(CallInput(transaction_data)), // SCALE-encoded parameters
-        )
-        .returns::<()>()
-        .fire()
-        .map_err(|_| Error::TransactionFailed);
-    result
-}
+    #[ink(storage)]
+    #[derive(Default)]
+    pub struct CallContract {}
+
+    #[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+    #[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+    pub enum Error {
+        TransactionFailed,
+    }
+    type Result<T> = core::result::Result<T, Error>;
+
+
+    impl CallContract{
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Default::default()
+        }
 
+        #[ink(message, payable)]
+        pub fn invoke_transaction(
+            &mut self,
+            callee: AccountId,
+            transfer_amount: u128,
+            function_selector: [u8; 4],
+            transaction_data: Vec<u8>,
+            gas_limit: Option<u64>,
+        ) -> Result<()> {
+        
+            let transaction_result = build_call::<<Self as ::ink::env::ContractEnv>::Env>()
+                .call_type(
+                    Call::new(callee) // contract to call
+                        .gas_limit(gas_limit.unwrap_or_default())
+                        .transferred_value(transfer_amount), // value to transfer with call
+                )
+                .exec_input(
+                    ExecutionInput::new(Selector::new(function_selector))
+                        .push_arg(transaction_data), // SCALE-encoded parameters
+                )
+                .returns::<()>()
+                .try_invoke();
+            
+            let result = match transaction_result {
+                Ok(Ok(_)) => Ok(()),
+                _ => Err(Error::TransactionFailed),
+            };
+
+            result
+        }
+    }
+}
 ```
 
 Note: the `function_selector` bytes can be found in the generated `target/ink/<contract-name>.json`.
 
-## Limitations of ink! v3
-
-- Multi-file projects are not supported with pure ink!
```

Comment:
This is still the case

Reply:
Thanks for the feedback, requested limitation note added back.